src/group_theory/perm/cycles.lean

```diff
@@ -617,6 +627,29 @@ begin
   { exact if_neg (mt same_cycle.apply_eq_self_iff (by tauto)) },
 end
 
+@[simp] lemma cycle_of_self_apply [fintype α] (f : perm α) (x : α) :
+  cycle_of f (f x) = cycle_of f x :=
+begin
+  rw (same_cycle.symm _).cycle_of_eq,
+  rw same_cycle_apply
+end
+
+@[simp] lemma cycle_of_self_apply_pow [fintype α] (f : perm α) (n : ℕ) (x : α) :
+  cycle_of f ((f ^ n) x) = cycle_of f x :=
+begin
+  rw same_cycle.cycle_of_eq,
+  rw same_cycle_pow_left_iff
+end
+
+@[simp] lemma cycle_of_self_apply_gpow [fintype α] (f : perm α) (n : ℤ) (x : α) :
+  cycle_of f ((f ^ n) x) = cycle_of f x :=
+begin
+  cases n,
+  { exact cycle_of_self_apply_pow _ _ _ },
+  { rw same_cycle.cycle_of_eq,
+    rw [gpow_neg_succ_of_nat, ←inv_pow, ←same_cycle_inv, same_cycle_pow_left_iff] }
+end
```

Comment:
Even more splitting apart: if you put this after `same_cycle_pow_left_iff`
```lean
@[simp] lemma same_cycle_gpow_left_iff {f : perm β} {x y : β} {n : ℤ} :
  same_cycle f ((f ^ n) x) y ↔ same_cycle f x y :=
begin
  cases n,
  { apply same_cycle_pow_left_iff },
  { rw [gpow_neg_succ_of_nat, ← inv_pow, ← same_cycle_inv, same_cycle_pow_left_iff,
        same_cycle_inv] },
end
```
then this can become
```suggestion
@[simp] lemma cycle_of_self_apply_gpow [fintype α] (f : perm α) (n : ℤ) (x : α) :
  cycle_of f ((f ^ n) x) = cycle_of f x :=
(same_cycle_gpow_left_iff.mpr (same_cycle.refl _ _)).cycle_of_eq
```
like the others.